### PR TITLE
Feature/save transfer draft

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Integrate the create transfer note request with the controller, with unit tests.
Added some initial code for retrieval of the transfer note and mapping of that data, will need to be expanded on in upcoming stories. 
Additional code to link a transfer note with the user selected categories.
The file change isnt as bad as it look as includes bit of refactoring.

https://dev.azure.com/civica-digital/EA%20-%20WEEE/_workitems/edit/44478

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/TransferTonnage.cshtml [hunk 1]
```diff
@@ -178,7 +179,10 @@
         </table>
 
         <div class="govuk-!-margin-top-3">
-            @this.WeeeGds().Button("Continue", new { id = "transfer-tonnage-button", @class = "govuk-!-width-one-quarter" })
+            <div class="govuk-button-group">
+                @*@this.WeeeGds().Button("Submit evidence note", new { name = "Action", value = ActionEnum.Submit, id = "submit-note", @class = "govuk-!-width-one-quarter" })*@
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/TransferTonnage.cshtml:183]: that seems commented out? still needed?

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.Database/scripts/Update/Release 5/Sprint05/20220516-1123-transfer-note-categories.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿CREATE TABLE [Evidence].[NoteTransferCategory](
+	[Id] [uniqueidentifier] NOT NULL,
+	[TransferNoteId] [uniqueidentifier] NOT NULL,
+	[CategoryId] [int] NOT NULL,
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Database/scripts/Update/Release 5/Sprint05/20220516-1123-transfer-note-categories.sql:4]: This table is only used to remember the categories that the user selects when creating an evidence note. I put it in as there could be a case where they select a list of categories, then on the enter tonnage screen they dont enter any values for a particular category. When later having to edit the category this table is used to remember their category selection as there isn't a way for us to reload the selected categories. *if that makes any sense at all*

File: src/EA.Weee.RequestHandlers/Mappings/TransferEvidenceNoteMap.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            this.mapper = mapper;
+        }
+
+        public TransferEvidenceNoteData Map(Note source)
+        {
+            return new TransferEvidenceNoteData
+            {
+                Id = source.Id,
+                Reference = source.Reference,
+                Type = (NoteType)source.NoteType.Value,
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/Mappings/TransferEvidenceNoteMap.cs:24]: This one is going the other way, Its going from Domain to Core. In theory could create a ToCoreEnumration, may look at doing that in another branch

File: src/EA.Weee.Database/scripts/Update/Release 5/Sprint05/20220516-1123-transfer-note-categories.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Database/scripts/Update/Release 5/Sprint05/20220516-1123-transfer-note-categories.sql:4]: okay, serves a purpose, fine

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                Status = source.TransferEvidenceNoteData.Status,
+                SchemeId = source.SchemeId
+            };
+
+            SetSuccessMessage(source.TransferEvidenceNoteData, source.DisplayNotification, model);
+
+            return model;
+        }
+
+        private void SetSuccessMessage(TransferEvidenceNoteData note, object displayMessage, ViewTransferNoteViewModel model)
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs:27]: displayMessage is bool display should handle the null check as casts it to false

File: src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/TransferTonnage.cshtml (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/TransferTonnage.cshtml:183]: Will be put back in for the submit story

**Pull request merged**